### PR TITLE
Refactor some PE/COFF unwinding classes to use clear interfaces

### PR DESCRIPTION
**Pull request by ronaldfw**
This is a mostly mechanical refactor to better support mocking for
tests for the unwinder that integrates all these parts (to be added
later).

Tested: Unit tests pass.
Bug: http://b/194768602

### CODE REVIEW

File: third_party/libunwindstack/PeCoffEpilog.h [hunk 1]
```diff
@@ -27,17 +27,39 @@
 
 namespace unwindstack {
 
-// Helper class for epilog detection and handling used for X86_64 unwinding for PE/COFF modules.
 class PeCoffEpilog {
  public:
-  PeCoffEpilog(Memory* file_memory, uint64_t text_section_vmaddr, uint64_t text_section_offset)
+  virtual ~PeCoffEpilog(){};
```
Comment (beckerhe) [inline third_party/libunwindstack/PeCoffEpilog.h:32]: Nit: The semicolon is superfluous

File: third_party/libunwindstack/PeCoffUnwindInfos.h [hunk 2]
```diff
@@ -68,21 +68,27 @@ struct UnwindInfo {
...
+ public:
+  explicit PeCoffUnwindInfosImpl(Memory* memory) : pe_coff_memory_(memory) {}
+  bool GetUnwindInfo(uint64_t unwind_info_file_offset, UnwindInfo* unwind_info) override;
+
+ private:
+  bool ParseUnwindInfoAtOffset(uint64_t file_offset, UnwindInfo* unwind_info);
+  PeCoffMemory pe_coff_memory_;
   // This is a cache of unwind infos.
   std::unordered_map<uint64_t, UnwindInfo> unwind_info_offset_to_unwind_info_;
 };
```
Comment (beckerhe) [inline third_party/libunwindstack/PeCoffUnwindInfos.h:94]: It's not really an implementation class if it is defined in a public header. Would exposing a factory function suffice in this case or do you really need the `Impl` class here?

Comment (beckerhe) [reply third_party/libunwindstack/PeCoffUnwindInfos.h:94]: (This comment also applies to the other Impl classes. As far as I've seen they are also declared in a public header.)

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: third_party/libunwindstack/PeCoffUnwindInfos.h (hunk 2 above)
Comment (ronaldfw) [reply third_party/libunwindstack/PeCoffUnwindInfos.h:94]: Yes, you are right, will add a factory function to not expose the Impl classes in the header.

Comment (ronaldfw) [reply third_party/libunwindstack/PeCoffUnwindInfos.h]: Ok, should be addressed now. Let me know if this looks alright.

**Pull request merged**